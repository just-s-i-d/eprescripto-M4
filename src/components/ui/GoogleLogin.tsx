import { Button, Divider, Image } from "antd";
import googleSvg from "@assets/google.svg";

type GoogleLoginProps = {
  host: string;
};
const GoogleLogin = ({ host }: GoogleLoginProps) => {
  return (
    <>
      <Divider className="w-9/12 min-w-9/12">Or</Divider>
      <div className="flex justify-center">
        <Button
          className="flex items-center gap-2 py-5 px-8 text-base bg-white"
          href={`${host}api/connect/google`}
        >
          <Image src={googleSvg} width={20} preview={false} alt="google" /> Sign
          in with Google
        </Button>
      </div>
    </>
  );
};
export default GoogleLogin;
