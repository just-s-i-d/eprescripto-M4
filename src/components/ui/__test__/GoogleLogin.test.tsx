import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import GoogleLogin from "../GoogleLogin";
import { MemoryRouter } from "react-router-dom";

describe("GoogleLogin Component", () => {
  const host = "http://localhost:1337/";
  it("renders correctly", () => {
    const { asFragment } = render(<GoogleLogin host={host} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders GoogleLogin component with a button", () => {
    render(<GoogleLogin host="host" />);
    expect(screen.getByText("Or")).toBeInTheDocument();
    expect(screen.getByText("Sign in with Google")).toBeInTheDocument();
  });

  it("button click triggers the correct URL", () => {
    render(
      <MemoryRouter>
        <GoogleLogin host={host} />
      </MemoryRouter>,
    );
    const btn = screen.getByRole("link");
    expect(btn).toHaveAttribute("href", `${host}api/connect/google`);
  });

  it("renders the Google SVG image", () => {
    render(<GoogleLogin host="host" />);
    expect(screen.getByAltText("google")).toBeInTheDocument();
  });
});
