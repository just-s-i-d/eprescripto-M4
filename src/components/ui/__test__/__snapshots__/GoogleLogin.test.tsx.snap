// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleLogin Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-divider css-dev-only-do-not-override-gzal6t ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center w-9/12 min-w-9/12"
    role="separator"
  >
    <span
      class="ant-divider-inner-text"
    >
      Or
    </span>
  </div>
  <div
    class="flex justify-center"
  >
    <a
      class="ant-btn css-dev-only-do-not-override-gzal6t ant-btn-default flex items-center gap-2 py-5 px-8 text-base bg-white"
      href="http://localhost:1337/api/connect/google"
      tabindex="0"
    >
      <div
        class="ant-image css-dev-only-do-not-override-gzal6t"
        style="width: 20px;"
      >
        <img
          alt="google"
          class="ant-image-img css-dev-only-do-not-override-gzal6t"
          src="google.svg"
          width="20"
        />
      </div>
      <span>
         Sign in with Google
      </span>
    </a>
  </div>
</DocumentFragment>
`;
